### PR TITLE
[DOC-4851] Document Private Service Connect for Dedicated clusters on GCP

### PR DESCRIPTION
**Pull request by mdlinville**
[DOC-4851] Document Private Service Connect for Dedicated clusters on GCP

### Previews
**Cloud release note**: [releases/cloud.md](https://deploy-preview-18550--cockroachdb-docs.netlify.app/docs/releases/cloud.html#2024-05-20
**24.1.0 feature highlight**: Not part of this PR, being drafted in https://docs.google.com/spreadsheets/d/1uufzdRz9PjD3hzDUVJV8piFBsjQM2XZcnhK7MYiBOd8/edit?disco=AAABM8eo7xo 

**Major**: 
[src/current/cockroachcloud/connect-to-your-cluster.md](https://deploy-preview-18550--cockroachdb-docs.netlify.app/docs/cockroachcloud/connect-to-your-cluster.html)
[src/current/cockroachcloud/network-authorization.md](https://deploy-preview-18550--cockroachdb-docs.netlify.app/docs/cockroachcloud/network-authorization.html)
[src/current/cockroachcloud/create-your-cluster.md](https://deploy-preview-18550--cockroachdb-docs.netlify.app/docs/cockroachcloud/create-your-cluster.html)

**Minor**:
[src/current/cockroachcloud/aws-privatelink.md](https://deploy-preview-18550--cockroachdb-docs.netlify.app/docs/cockroachcloud/aws-privatelink.html)
[src/current/cockroachcloud/security-overview.md](https://deploy-preview-18550--cockroachdb-docs.netlify.app/docs/cockroachcloud/security-overview.html)
[src/current/v23.1/security-reference/security-overview.md](https://deploy-preview-18550--cockroachdb-docs.netlify.app/docs/v23.1/security-reference/security-overview.html)
[src/current/v23.2/security-reference/security-overview.md](https://deploy-preview-18550--cockroachdb-docs.netlify.app/docs/v23.2/security-reference/security-overview.html)
[src/current/v24.1/security-reference/security-overview.md](https://deploy-preview-18550--cockroachdb-docs.netlify.app/docs/v24.1/security-reference/security-overview.html)

### CODE REVIEW

File: src/current/cockroachcloud/connect-to-your-cluster.md [hunk 1]
```diff
@@ -21,7 +21,7 @@ By default, CockroachDB {{ site.data.products.dedicated }} clusters are locked d
 
 - Allowed IP address ranges on the internet.
 - Cloud-provider-specific peer networking options:
-    - Google Cloud Platform (GCP) VPC Peering
+    - Google Cloud Platform (GCP) VPC Peering or Private Service Connect (Preview)
     - Amazon Web Services (AWS) Private link
```
Comment (devarshishah3) [inline src/current/cockroachcloud/connect-to-your-cluster.md:25]: nit:  AWS PrivateLink

File: src/current/cockroachcloud/connect-to-your-cluster.md [hunk 2]
```diff
@@ -36,22 +36,53 @@ Removing or adding an authorized network on your CockroachDB {{ site.data.produc
...
+AWS PrivateLink can be configured only after the cluster is created. For detailed instructions, refer to [Managing AWS PrivateLink for a cluster]({% link cockroachcloud/aws-privatelink.md %}).
+{{site.data.alerts.end}}
 
 Azure Private Link is not yet available for [CockroachDB {{ site.data.products.dedicated }} on Azure]({% link cockroachcloud/cockroachdb-dedicated-on-azure.md %}).
 
 {{site.data.alerts.callout_info}}
 {% include cockroachcloud/cdc/kafka-vpc-limitation.md %}
 {{site.data.alerts.end}}
 
+#### Private Service Connect
```
Comment (devarshishah3) [inline src/current/cockroachcloud/connect-to-your-cluster.md:59]: prepend GCP

File: src/current/cockroachcloud/create-your-cluster.md [hunk 3]
```diff
@@ -60,9 +60,17 @@ Click **Next: Capacity**.
...
+You can use [VPC peering]({% link cockroachcloud/network-authorization.md %}#vpc-peering) to establish a private connection between a GCP application and a CockroachDB {{ site.data.products.dedicated }} cluster deployed on GCP. A separate VPC Peering connection is required for each cluster.
 
-VPC peering is only available for GCP clusters. For clusters deployed on AWS, you can [set up AWS PrivateLink]({% link cockroachcloud/network-authorization.md %}#aws-privatelink) after creating your cluster. [Azure Virtual Network Peering](https://learn.microsoft.com/azure/virtual-network/virtual-network-peering-overview) is not yet supported. Refer to [CockroachDB {{ site.data.products.dedicated }} on Azure]({% link cockroachcloud/cockroachdb-dedicated-on-azure.md %}).
+{{site.data.alerts.callout_success}}
+You can [set up Private Service Connect (Preview)]({% link cockroachcloud/network-authorization.md %}#private-service-connect) instead of VPC peering after creating your cluster.
```
Comment (devarshishah3) [inline src/current/cockroachcloud/create-your-cluster.md:66]: prepend GCP

File: src/current/cockroachcloud/network-authorization.md [hunk 4]
```diff
@@ -72,16 +73,22 @@ Refer to:
 - [Connect to a CockroachDB {{ site.data.products.serverless }} Cluster: Authorize your network]({% link cockroachcloud/connect-to-a-serverless-cluster.md %}#authorize-your-network).
 - [Connect to a CockroachDB {{ site.data.products.dedicated }} Cluster: Authorize your network]({% link cockroachcloud/connect-to-your-cluster.md %}#authorize-your-network).
 
+## Private Service Connect
```
Comment (devarshishah3) [inline src/current/cockroachcloud/network-authorization.md:76]: prepend GCP

File: src/current/cockroachcloud/create-your-cluster.md [hunk 5]
```diff
@@ -60,9 +60,17 @@ Click **Next: Capacity**.
...
+{{site.data.alerts.end}}
+
+{{site.data.alerts.callout_success}}
+For clusters deployed on AWS, you can [set up AWS PrivateLink]({% link cockroachcloud/network-authorization.md %}#aws-privatelink) after creating your cluster.
+{{site.data.alerts.end}}
+
+[Azure Virtual Network Peering](https://learn.microsoft.com/azure/virtual-network/virtual-network-peering-overview) is not yet supported. Refer to [CockroachDB {{ site.data.products.dedicated }} on Azure]({% link cockroachcloud/cockroachdb-dedicated-on-azure.md %}).
 
 You can use CockroachDB {{ site.data.products.cloud }}'s default IP range and size (`172.28.0.0/14`) as long as it doesn't overlap with the IP ranges in your network. Alternatively, you can configure the IP range:
```
Comment (devarshishah3) [inline src/current/cockroachcloud/create-your-cluster.md:76]: Reword Step 3 to include GCP Private Service Connect. Currently, it only has VPC Peering or AWS PrivateLink
image

File: src/current/cockroachcloud/connect-to-your-cluster.md [hunk 6]
```diff
@@ -36,22 +36,53 @@ Removing or adding an authorized network on your CockroachDB {{ site.data.produc
...
+
+{{site.data.alerts.callout_success}}
+After validation succeeds for an endpoint, additional endpoints in the same VPC are automatically automatically accepted if they are configured with the cluster's target service ID. Additional VPCs must be added separately.
+{{site.data.alerts.end}}
+
+##### Known limitations
+
+Deleting a PSC connection from CockroachDB {{ site.data.products.cloud }} is not supported. Instead, remove or disable the connection from GCP or modify its target service ID so that it no longer matches the cluster. When a connection no longer exists, CockroachDB {{ site.data.products.cloud }} automatically removes it from the cluster.
+
 #### VPC Peering
```
Comment (devarshishah3) [inline src/current/cockroachcloud/connect-to-your-cluster.md:86]: Shouldn't VPC peering be one heading highere? ### VPC Peering

Comment (mdlinville) [reply src/current/cockroachcloud/connect-to-your-cluster.md:86]: It is at level 4, the same level as PSC.

File: src/current/v24.1/security-reference/security-overview.md [hunk 7]
```diff
@@ -169,7 +169,7 @@ CockroachDB {{ site.data.products.core }} here refers to the situation of a user
   <td>✓</td>
   <td>✓</td>
   <td>✓</td>
-  <td><a href="https://www.cockroachlabs.com/docs/cockroachcloud/create-your-cluster.html#step-8-enable-vpc-peering-optional">VPC Peering</a> for GCP clusters and <a href="https://www.cockroachlabs.com/docs/cockroachcloud/aws-privatelink">AWS PrivateLink</a> for AWS clusters </td>
+  <td><a href="https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster.html#private-service-connect">Private Service Connect (PSC) (Preview)</a<a href="https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster.html#vpc-peering">VPC Peering</a> for GCP clusters and <a href="https://www.cockroachlabs.com/docs/cockroachcloud/aws-privatelink">AWS PrivateLink</a> for AWS clusters </td>
```
[the remark this answers is not in the document]
Comment (mdlinville) [reply src/current/v24.1/security-reference/security-overview.md:172]: There is a problem with the area where you left this comment, but the screenshot you provided is showing a different bug on the Cloud page, I think.

**Review by devarshishah3 — APPROVED** [review]

File: src/current/_includes/releases/cloud/2024-05-20.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+## May 20, 2024
+
+<h3 id="2024-05-20-security-updates"> Security updates </h3>
+
+- [Configuring private connectivity using Google Cloud Private Service Connect]({% link cockroachcloud/connect-to-your-cluster.md %}#gcp-private-service-connect) is available in [preview](https://www.cockroachlabs.com/docs/{{site.current_cloud_version}}/cockroachdb-feature-availability) for CockroachDB {{ site.data.products.dedicated }} clusters on GCP. [Private connectivity]({% link cockroachcloud/network-authorization.md %}#options-for-controlling-network-access) allows you to establish SQL access to a CockroachDB {{ site.data.products.dedicated }} cluster entirely through cloud provider private infrastructure, without exposing the cluster to the public internet, affording enhanced security and performance.
```
Comment (mdlinville) [inline src/current/_includes/releases/cloud/2024-05-20.md:5]: @devarshishah3 PTAL, this is modeled on the note we used for AWS PrivateLink on multi-region Serverless.

File: src/current/cockroachcloud/network-authorization.md [hunk 9]
```diff
@@ -72,29 +76,6 @@ Refer to:
 - [Connect to a CockroachDB {{ site.data.products.serverless }} Cluster: Authorize your network]({% link cockroachcloud/connect-to-a-serverless-cluster.md %}#authorize-your-network).
 - [Connect to a CockroachDB {{ site.data.products.dedicated }} Cluster: Authorize your network]({% link cockroachcloud/connect-to-your-cluster.md %}#authorize-your-network).
 
-## VPC peering
```
Comment (mdlinville) [inline src/current/cockroachcloud/network-authorization.md:75]: The most important bits of this section are moved to `connect-to-your-cluster.md`.

File: src/current/cockroachcloud/security-overview.md [hunk 10]
```diff
@@ -129,12 +129,17 @@ The following table summarizes the CockroachDB {{ site.data.products.cloud }} se
...
+    <td><a href="{% link cockroachcloud/network-authorization.md %}#gcp-private-service-connect">Private Service Connect (PSC) (Preview)</a> for GCP clusters</td>
   </tr>
   <tr>
     <td>&nbsp;</td>
     <td>✓</td>
-    <td><a href="{% link cockroachcloud/network-authorization.md %}#aws-privatelink"></a>PrivateLink for AWS clusters </td>
+    <td><a href="{% link cockroachcloud/network-authorization.md %}#gcp-vpc-peering">VPC Peering</a> for GCP clusters</td>
+  </tr>
+  <tr>
+    <td>✓<a href="#privatelink-multiregion-serverless"><sup>1</sup></a></td>
```
Comment (mdlinville) [inline src/current/cockroachcloud/security-overview.md:140]: This and the below were misses when PrivateLink for Serverless launched.

File: src/current/v23.1/security-reference/security-overview.md [hunk 11]
```diff
@@ -165,11 +165,11 @@ CockroachDB {{ site.data.products.core }} here refers to the situation of a user
    <td>Network-level Configuration of allowed IP addresses</td>
  </tr>
  <tr>
-  <td>&nbsp;</td>
+  <td>✓<a href="#privatelink-multiregion-serverless"><sup>1</sup></a></td>
```
Comment (mdlinville) [inline src/current/v23.1/security-reference/security-overview.md:168]: This and the below were misses when PrivateLink for Serverless launched. This table is a little different from the Cloud one.

**Review by mdlinville — COMMENTED** [review]
Hints for @rmloveland review :)

File: src/current/cockroachcloud/connect-to-your-cluster.md [hunk 12]
```diff
@@ -21,8 +21,8 @@ By default, CockroachDB {{ site.data.products.dedicated }} clusters are locked d
 
 - Allowed IP address ranges on the internet.
 - Cloud-provider-specific peer networking options:
-    - Google Cloud Platform (GCP) VPC Peering
-    - Amazon Web Services (AWS) Private link
+    - Google Cloud Platform (GCP) VPC Peering or Private Service Connect (Preview)
```
Comment (rmloveland) [inline src/current/cockroachcloud/connect-to-your-cluster.md:24]: later on I'm seeing there are anchors these could link to: `#gcp-private-service-connect` etc

File: src/current/cockroachcloud/connect-to-your-cluster.md [hunk 13]
```diff
@@ -36,23 +36,57 @@ Removing or adding an authorized network on your CockroachDB {{ site.data.produc
...
 {{site.data.alerts.end}}
 
-#### VPC Peering
+#### GCP Private Service Connect
+
+{{site.data.alerts.callout_info}}
+{% include_cached feature-phases/preview.md %}
+{{site.data.alerts.end}}
+
+1. Navigate to your cluster's **Networking > Private endpoint** tab.
```
Comment (rmloveland) [inline src/current/cockroachcloud/connect-to-your-cluster.md:65]: not for this PR thoughts:

as a user I'd love these "your cluster's FOO tab" mentions to all be links to CC console things

maybe not possible unless we have some kind of SSO across docs site / console

one day maybe we'll ship these docs IN console and solve this

File: src/current/cockroachcloud/network-authorization.md [hunk 14]
```diff
@@ -16,19 +16,23 @@ You can authorize network access to your cluster by:
...
 
-    For CockroachDB {{ site.data.products.dedicated }} clusters deployed on GCP, refer to [Google Cloud Platform (GCP) Virtual Private Cloud (VPC) peering](#vpc-peering). For CockroachDB {{ site.data.products.dedicated }} clusters or multi-region CockroachDB {{ site.data.products.serverless }} clusters deployed on AWS, refer to [Amazon Web Service (AWS) PrivateLink](#aws-privatelink).
+    - <a id="gcp-private-service-connect"></a><a id="gcp-vpc-peering"></a><a id="vpc-peering"></a>CockroachDB {{ site.data.products.dedicated }} clusters deployed on GCP can connect privately using GCP Private Service Connect (PSC) (Preview) or GCP VPC peering. PSC allows you to selectively connect your cluster to a VPC within your Google Cloud project, while VPC Peering allows you to peer your cluster's VPC managed by Cockroach Cloud's VPC with a VPC within your Google Cloud project.
```
Comment (rmloveland) [inline src/current/cockroachcloud/network-authorization.md:19]: these mentions of GCP private connect and VPC peering could link to the named sections on the 'connect to your cluster' page for user convenience

File: src/current/cockroachcloud/security-overview.md [hunk 15]
```diff
@@ -154,3 +159,5 @@ The following table summarizes the CockroachDB {{ site.data.products.cloud }} se
     <td>CockroachDB, as a distributed SQL database, is uniquely resilient by nature. A cluster can tolerate node failures as long as the majority of nodes remain functional. See <a href="https://www.cockroachlabs.com/docs/{{ site.current_cloud_version }}/demo-fault-tolerance-and-recovery">Disaster Recovery.</a></td>
   </tr>
 </table>
+
+<a id="privatelink-multiregion-serverless">1</a>: AWS PrivateLink is in preview for multi-region Serverless clusters, and is not supported for single-region Serverless clusters. Refer to <a href="https://www.cockroachlabs.com/docs/cockroachcloud/aws-privatelink?filters=serverless">Manage AWS PrivateLink</a>.
```
Comment (rmloveland) [inline src/current/cockroachcloud/security-overview.md:163]: 'preview' mention could link to our definition of what 'preview' means

File: src/current/v23.1/security-reference/security-overview.md [hunk 16]
```diff
@@ -165,11 +165,11 @@ CockroachDB {{ site.data.products.core }} here refers to the situation of a user
...
   <td>✓</td>
   <td>✓</td>
   <td>✓</td>
-  <td><a href="https://www.cockroachlabs.com/docs/cockroachcloud/create-your-cluster.html#step-8-enable-vpc-peering-optional">VPC Peering</a> for GCP clusters and <a href="https://www.cockroachlabs.com/docs/cockroachcloud/aws-privatelink">AWS PrivateLink</a> for AWS clusters </td>
+  <td><a href="https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster.html#private-service-connect">Private Service Connect (PSC) (Preview)</a> or <a href="https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster.html#vpc-peering">VPC Peering</a> for GCP clusters and <a href="https://www.cockroachlabs.com/docs/cockroachcloud/aws-privatelink">AWS PrivateLink</a> for AWS clusters </td>
```
Comment (rmloveland) [inline src/current/v23.1/security-reference/security-overview.md:172]: I believe this syntax can be `{% link cockroachcloud/connect-to-your-cluster.md %}#the-anchor` however writing that out is such a faff and doesn't save many characters anyway so I leave it to your conscience / perfectionism or lack thereof (or how much of a rush you're in)

File: src/current/v24.1/security-reference/security-overview.md [hunk 17]
```diff
@@ -165,11 +165,11 @@ CockroachDB {{ site.data.products.core }} here refers to the situation of a user
    <td>Network-level Configuration of allowed IP addresses</td>
  </tr>
  <tr>
-  <td>&nbsp;</td>
+  <td>✓<a href="#privatelink-multiregion-serverless"><sup>1</sup></a></td>
   <td>✓</td>
   <td>✓</td>
   <td>✓</td>
-  <td><a href="https://www.cockroachlabs.com/docs/cockroachcloud/create-your-cluster.html#step-8-enable-vpc-peering-optional">VPC Peering</a> for GCP clusters and <a href="https://www.cockroachlabs.com/docs/cockroachcloud/aws-privatelink">AWS PrivateLink</a> for AWS clusters </td>
+  <td><a href="https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster.html#gcp-private-service-connect">GCP Private Service Connect (PSC) (Preview)</a> or <a href="https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster.html#vpc-peering">VPC Peering</a> for GCP clusters and <a href="https://www.cockroachlabs.com/docs/cockroachcloud/aws-privatelink">AWS PrivateLink</a> for AWS clusters </td>
```
Comment (rmloveland) [inline src/current/v24.1/security-reference/security-overview.md:172]: same comments as above re: link syntax and link to definition of 'preview'

**Review by rmloveland — APPROVED** [review]
LGTM!

File: src/current/cockroachcloud/security-overview.md [hunk 18]
```diff
@@ -129,12 +129,17 @@ The following table summarizes the CockroachDB {{ site.data.products.cloud }} se
   <tr>
     <td>&nbsp;</td>
     <td>✓</td>
-    <td><a href="{% link cockroachcloud/network-authorization.md %}#vpc-peering">VPC Peering</a> for GCP clusters</td>
+    <td><a href="{% link cockroachcloud/network-authorization.md %}#gcp-private-service-connect">Private Service Connect (PSC) (Preview)</a> for GCP clusters</td>
```
[the remark this answers is not in the document]
Comment (mdlinville) [reply src/current/cockroachcloud/security-overview.md:132]: In Cloud, Preview means you can try it self-service but things might change. Private preview and Limited Access you need to be enrolled first. Good idea about removing it from these links.

**Pull request merged**